xarray/core/common.py

```diff
@@ -761,12 +759,12 @@ def resample(self, indexer=None, skipna=None, closed=None, label=None,
                 'errors.'
             )
 
-        group = DataArray(dim_coord, coords=dim_coord.coords,
-                          dims=dim_coord.dims, name=RESAMPLE_DIM)
+        group = IndexVariable(
+            data=self.indexes[dim], dims=[dim], name=RESAMPLE_DIM)
```

Comment:
I *think* we could alternatively fix this by replace `dim_coord` in the `data` argument with `dim_coord.to_index()`, e.g., compare:
```
ipdb> DataArray(dim_coord, coords=dim_coord.coords, dims=dim_coord.dims, name=RESAMPLE_DIM).to_index()
Index([2000-01-01 00:00, 2000-01-01 06:00, 2000-01-01 12:00, 2000-01-01 18:00,
       2000-01-02 00:00, 2000-01-02 06:00, 2000-01-02 12:00, 2000-01-02 18:00,
       2000-01-03 00:00, 2000-01-03 06:00],
      dtype='object', name='time')
ipdb> DataArray(dim_coord.to_index(), coords=dim_coord.coords, dims=dim_coord.dims, name=RESAMPLE_DIM).to_index()
PeriodIndex(['2000-01-01 00:00', '2000-01-01 06:00', '2000-01-01 12:00',
             '2000-01-01 18:00', '2000-01-02 00:00', '2000-01-02 06:00',
             '2000-01-02 12:00', '2000-01-02 18:00', '2000-01-03 00:00',
             '2000-01-03 06:00'],
            dtype='period[6H]', name='time', freq='6H')
```

We go to some effort to preserve pandas.Index classes when passed into DataArray/Variable but apparently don't do it consistently everywhere. So there's also probably a lower level fix somewhere (perhaps in `xarray.core.variable.as_compatible_data`?) to ensure that this happens more consistently.

Reply:
Great! I'll have a look